mcs/class/System/Mono.UnityTls/UnityTls.cs

```diff
@@ -0,0 +1,257 @@
+            public unitytls_tlsctx_set_certificate_callback_t             unitytls_tlsctx_set_certificate_callback;
+            public delegate void                                          unitytls_tlsctx_set_trace_callback_t(unitytls_tlsctx* ctx, unitytls_tlsctx_trace_callback cb, void* userData, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_set_trace_callback_t                   unitytls_tlsctx_set_trace_callback;
+            public delegate void                                          unitytls_tlsctx_set_x509verify_callback_t(unitytls_tlsctx* ctx, unitytls_tlsctx_x509verify_callback cb, void* userData, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_set_x509verify_callback_t              unitytls_tlsctx_set_x509verify_callback;
+            public delegate void                                          unitytls_tlsctx_set_supported_ciphersuites_t(unitytls_tlsctx* ctx, unitytls_ciphersuite* supportedCiphersuites, size_t supportedCiphersuitesLen, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_set_supported_ciphersuites_t           unitytls_tlsctx_set_supported_ciphersuites;
+            public delegate unitytls_ciphersuite                          unitytls_tlsctx_get_ciphersuite_t(unitytls_tlsctx* ctx, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_get_ciphersuite_t                      unitytls_tlsctx_get_ciphersuite;
+            public delegate unitytls_protocol                             unitytls_tlsctx_get_protocol_t(unitytls_tlsctx* ctx, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_get_protocol_t                         unitytls_tlsctx_get_protocol;
+            public delegate unitytls_x509verify_result                    unitytls_tlsctx_process_handshake_t(unitytls_tlsctx* ctx, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_process_handshake_t                    unitytls_tlsctx_process_handshake;
+            public delegate size_t                                        unitytls_tlsctx_read_t(unitytls_tlsctx* ctx, UInt8* buffer, size_t bufferLen, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_read_t                                 unitytls_tlsctx_read;
+            public delegate size_t                                        unitytls_tlsctx_write_t(unitytls_tlsctx* ctx, UInt8* data, size_t bufferLen, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_write_t                                unitytls_tlsctx_write;
+            public delegate void                                          unitytls_tlsctx_free_t(unitytls_tlsctx* ctx);
+            public unitytls_tlsctx_free_t                                 unitytls_tlsctx_free;
+        }
+
+        [DllImport("__Internal")]
+        private static extern IntPtr mono_unity_get_unitytls_interface();
```

Comment:
This should likely be an internal call and connected via the normal mono icall mechanism for class libraries.

Reply:
As is, we need to be sure to export this symbol across all platforms which is not always easy.

Reply:
Switched it to icall: 65af3e711c50cd6547b3fcf5a9166bafa3890e5c
Completely fresh build(.zip) from katana still passes my tests so I guess I did everything right there? :)